h2o-extensions/xgboost/src/test/java/hex/tree/xgboost/XGBoostModelTest.java

```diff
@@ -9,6 +9,14 @@
 
 public class XGBoostModelTest {
   
+  @Test
+  public void shouldGetCorrectMaxThread() {
+    assertEquals(Runtime.getRuntime().availableProcessors(), XGBoostModel.getMaxNThread());
+    System.setProperty("sys.ai.h2o.xgboost.nthreadMax", "2");
```

Comment:
There is a dedicated JUnit library named [System rules](https://stefanbirkner.github.io/system-rules/) which will help you avoid all the problems associated with testing based on system properties. We already have the dependency (which you can update optionally) and it is already used in H2O.

If nothing else, it makes sure the environment variables are restored properly. See `SQLManagerTest` for example.

Reply:
thanks, used it